src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -127,26 +127,10 @@ final case class ClientApiServiceImpl(
     val operationLabel: String = s"Retrieving client $clientId"
     logger.info(operationLabel)
 
-    def isConsumerOrProducer(organizationId: UUID, client: PersistentClient)(implicit
-      ec: ExecutionContext
-    ): Future[Boolean] = {
-      def isProducer(): Future[Boolean] = Future
-        .traverse(client.purposes.map(_.eService.eServiceId))(catalogManagementService.getEServiceById)
-        .map(eservices => eservices.map(_.producerId).contains(organizationId))
-
-      if (client.consumerId == organizationId) Future.successful(true) else isProducer()
-
-    }
-
     val result: Future[Client] = for {
-      requesterUuid        <- getOrganizationIdFutureUUID(contexts)
-      clientUuid           <- clientId.toFutureUUID
-      client               <- authorizationManagementService.getClient(clientUuid)
-      isConsumerOrProducer <- isConsumerOrProducer(requesterUuid, client)
-      _                    <- Future
-        .failed(OrganizationNotAllowedOnClient(client.id.toString, requesterUuid))
-        .unlessA(isConsumerOrProducer)
-
+      requesterUuid <- getOrganizationIdFutureUUID(contexts)
+      clientUuid    <- clientId.toFutureUUID
+      client        <- authorizationManagementService.getClient(clientUuid)
     } yield client.toApi(requesterUuid == client.consumerId)
```

Comment:
The implementation is missing the part that manages `relationshipIds` inside the client.
The requirement is:
- if `organizationId` is **not the consumer** then `relationshipIds` field must be set to `List.empty`
- else retrieve the client unchanged

Reply:
you're right, I forgot it was already implemented 😬

Reply:
Same, sorry!